templates/cc/register.go

```diff
@@ -282,6 +286,12 @@ func cTypeOfString(s string) string {
 	case "uint64":
 		return "uint64_t"
 	default:
+		if strings.HasPrefix(s, "*") {
```

Comment:
I'd rather we caught these cases in the callers of cTypeOfString instead of kludging it like this, something like the `if t.IsEmbed()` above. A TODO for expediency is fine, but this feels kind of ugly.